ios/Video/RCTVideo.m

```diff
@@ -2064,4 +2100,67 @@ - (void)pictureInPictureController:(AVPictureInPictureController *)pictureInPict
 }
 #endif
 
+- (void)setAdOptions:(NSDictionary *)adOptions {
+  if ([adOptions valueForKey:@"adTagUrl"] != nil) {
+    self->adTagUrl = [adOptions valueForKey:@"adTagUrl"];
+  }
+}
+
+#pragma mark - IMAAdsLoaderDelegate
+
+- (void)adsLoader:(IMAAdsLoader *)loader adsLoadedWithData:(IMAAdsLoadedData *)adsLoadedData {
+  // Initialize and listen to the ads manager loaded for this request.
+  self.adsManager = adsLoadedData.adsManager;
+  
+  self.adsManager.delegate = self;
+  
+  [self.adsManager initializeWithAdsRenderingSettings:nil];
+}
+
+- (void)adsLoader:(IMAAdsLoader *)loader failedWithErrorData:(IMAAdLoadingErrorData *)adErrorData {
+  [self->_player play];
+}
+
+
+- (void)adsManager:(IMAAdsManager *)adsManager didReceiveAdEvent:(IMAAdEvent *)event {
+  // Play each ad once it has loaded.
+  if (event.type == kIMAAdEvent_LOADED) {
+    [adsManager start];
+  }
+   
+  if (self.onVideoAdEvent) {
+    // Make the eventname fully capsed with underscores instead of spaces.
+    // This makes event names consistent between iOS and Android
+    NSString *eventName = event.typeString.uppercaseString;
+    eventName = [eventName stringByReplacingOccurrencesOfString:@" " withString:@"_"];
+    
+    self.onVideoAdEvent(@{
+                           @"name": eventName,
+                           @"target": self.reactTag,
```

Comment:
events always have target and tag in them in the native events. I have no idea what the difference is since both of the values are the same, but maybe consider to add a tag key?
No particular preference for this tho, just wanted to point it out